### PR TITLE
Add NONE pooling mode to support pre-pooled model outputs

### LINKED ISSUE #4708: [BUG] MEAN pooling applied by default when no pooling should be used for pre-pooled model outputs
**What is the bug?**
The OpenSearch documentation states that sentence-transformer models traced with `sentence-transformers` library include post-processing (pooling), implying no additional pooling is needed. However, the code in `ONNXSentenceTransformerTextEmbeddingTranslator` always uses the first output (token_embeddings) and applies MEAN pooling by default (line 44), even for models that already provide pre-pooled `sentence_embedding` output.

This means models with pre-computed sentence embeddings get redundant MEAN pooling applied on top, causing:
- Incorrect embeddings that don't match Python inference output
- Redundant computation

Additionally, there is no `NONE` pooling mode available in the `PoolingMode` enum, so users have no way to explicitly skip pooling for pre-pooled models.

**How can one reproduce the bug?**
1. Export a sentence-transformers model (e.g., `all-MiniLM-L6-v2`) that includes pooling in its ONNX graph
2. Register the model in OpenSearch without specifying `pooling_mode`
3. Generate embeddings and compare with Python `sentence-transformers` output
4. Observe that OpenSearch embeddings differ from Python due to redundant MEAN pooling

**What is the expected behavior?**
Users should be able to specify `pooling_mode: "none"` to use pre-pooled model outputs directly, skipping any additional pooling computation. This ensures embeddings match the original model's Python inference output.

**What is your host/environment?**
- OpenSearch 3.x
- ml-commons plugin

**Do you have any additional context?**
The fix involves:
1. Adding `NONE("none")` to the `PoolingMode` enum in `BaseModelConfig`
2. Updating `ONNXSentenceTransformerTextEmbeddingTranslator` to use the second output (sentence_embedding) when pooling mode is NONE
3. Updating `HuggingfaceTextEmbeddingTranslator` to support NONE pooling with fallback logic for various output formats (`sentence_embedding` → `pooler_output` → positional fallback)

### PR DESCRIPTION
**Pull request by aneesh-db**
### Description
Adds `NONE` as a pooling option for text embedding models, fixing a bug where MEAN pooling was applied by default even for models that already provide pre-pooled `sentence_embedding` output.

The documentation states that sentence-transformer models traced with the sentence-transformers library include post-processing (pooling), implying no additional pooling is needed. However, `ONNXSentenceTransformerTextEmbeddingTranslator` always uses the first output (token_embeddings) and applies MEAN pooling by default, causing incorrect embeddings that don't match Python inference output.

With `pooling_mode: "none"`, models with pre-computed sentence embeddings can use those outputs directly without redundant pooling computation.

**Changes:**
- Add `NONE("none")` to `PoolingMode` enum in `BaseModelConfig`
- Update `ONNXSentenceTransformerTextEmbeddingTranslator` to use the second output (`sentence_embedding`) when pooling mode is NONE
- Update `HuggingfaceTextEmbeddingTranslator` to support NONE pooling with fallback logic (`sentence_embedding` → `pooler_output` → positional fallback)
- Add unit tests for NONE pooling in both ONNX and TorchScript
- Update documentation and release notes

### Related Issues
Resolves #4708

### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/12075).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-ml-commons.release-notes-3.4.0.0.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Compatible with OpenSearch and OpenSearch Dashboards version 3.4.0
 * allow higher maximum number of batch inference job tasks ([#4474](https://github.com/opensearch-project/ml-commons/pull/4474))
 
 ### Bug Fixes
+* Add NONE pooling mode to support pre-pooled model outputs, fixing bug where MEAN pooling was applied by default ([#4708](https://github.com/opensearch-project/ml-commons/issues/4708))
```
Comment (rithinpullela) [inline release-notes/opensearch-ml-commons.release-notes-3.4.0.0.md:12]: no need to add the release notes, can remove this

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/HuggingfaceTextEmbeddingTranslator.java [hunk 2]
```diff
@@ -78,9 +78,29 @@ public NDList processInput(TranslatorContext ctx, String input) {
...
+                embeddings = list.get("pooler_output");
+            }
+            if (embeddings == null && list.size() > 1) {
+                // Use second output if available
+                embeddings = list.get(1);
+            }
+            if (embeddings == null) {
+                // Fallback to first output
+                embeddings = list.get(0);
+            }
```
[the remark this answers is not in the document]
Comment (aneesh-db) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/HuggingfaceTextEmbeddingTranslator.java:97]: addressed in https://github.com/opensearch-project/ml-commons/pull/4744
